### DISCUSSION TITLE
loramac-node frame counter

### DISCUSSION THREAD

**Discussion by faridm23** [Lora-net/LoRaMac-node#1077] [Q&A]
I need to run a test where I want to send uplink messages with the same frame counter. Where in the stack can I find this fcnt byte that gets incremented whenever there is an uplink message?

**Comment by MedadRufus** [comment]
There is a function to get frame counter LoRaMacCryptoGetFCntUp(...)


https://github.com/Lora-net/LoRaMac-node/blob/3d940d323f80c093d54c118ca978cc7730ee5ec5/src/mac/LoRaMacCrypto.c#L843-L853

You could quite feasibly make function that sets FCntUp in a similar way.

**Reply by faridm23** [reply]
Thank you for your reply! yes I saw that. I tried to hard code the frame counter to a value of 1. But it seems like the frame counter kept incrementing as it should.

Here is what I did: 

    *currentUp = 1; // CryptoNvm->FCntList.FCntUp + 1;

**Reply by faridm23** [reply]
the answer is in this line for anyone who might have same question as I did: CryptoNvm->FCntList.FCntUp = "Fcntvalue";
